test/JsonClientTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+use GuzzleHttp\Client;
+use Leverage\Client\JsonClient;
+use Leverage\Client\JsonRequest;
+use Leverage\Encoder\JsonEncoder;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class JsonClientTest extends TestCase
+{
+    private JsonClient $client;
+
+    private MockObject & Client $delegate;
+    private MockObject & JsonEncoder $json;
+    private MockObject & RequestInterface $request;
+    private MockObject & ResponseInterface $response;
+
+    public function setUp(): void
+    {
+        $this->mockDelegate();
+        $this->json = self::createMock(JsonEncoder::class);
+        $this->request = self::createMock(JsonRequest::class);
+        $this->client = new JsonClient(
+            $this->delegate,
+            $this->json,
```

Comment:
I guess we can(?) agree to use named args everywhere.

Reply:
I agree for most cases. Eg I really like it for the case above.

I think there's a line to look out for.

Extreme example -

```$foo = $this->foos->findById(
    id: $id,
);```

I'm not sure whether it's a case of "single arguments don't require names" or "contextual method names don't require param names" or ... ?

Reply:
Oh right. Probably no need to use it for most of single arguments:
```
$foo->get($id);
$foo->findByAccount($account);
$foo->equalsTo($bar);
```